lib/string/README

```diff
@@ -0,0 +1,272 @@
+General guidelines:
+===================
+
+-  If there's an upper-case macro that wraps a function, use the macro
+   if possible.  These use macro magic to add safety.
+
+-  x*() functions wrap a function of the same name without the 'x'.
+   These wrappers exit on error instead of returning NULL.
+   They simplify program code.  Avoid them in library code.
+
+-  strn*() functions are forbidden for use with strings.  Consider strn
+   a mnemonic for [[gnu::nonstring]].  We use them with utmp(5) members
+   (almost) exclusively.
+
+
+Don't use some libc functions without Really Good Reasons:
+==========================================================
+
+<stdio.h>
+    asprintf(3)
+	Use aprintf() instead.
+	It is difficult to handle errors after asprintf(3).
+	Also, it makes it more difficult for static analyzers to check
+	that memory is later free(3)d appropriately.
+
+<string.h>
+    snprintf(3)
+	Use stprintf() or seprintf() instead.
+	It is difficult to check for truncation after snprintf(3).
+
+    strcmp(3)
+	Use streq() instead.
+	The return value of strcmp(3) is confusing.
+	strcmp(3) would be legitimate for sorting strings.
+
+    strcasecmp(3)
+	Use strcaseeq() instead.
+
+    strncat(3)
+	Use strndup(3) or strndupa(3) instead.
+	strncat(3) is legitimate for catenating a prefix of a string
+	after an existing string.
+
+    strlcpy(3)
+	Use strtcpy() instead.
+	strlcpy(3) is vulnerable to DoS.
+	Also, it is difficult to check for truncation after strlcpy(3).
+
+    strlcat(3)
+	Use strtcat() or stpecpy() instead.
+	strlcat(3) is vulnerable to DoS.
+	Also, it is difficult to check for truncation after strlcat(3).
+
+
+Specific guidelines:
+====================
+
+ctype/ - Character classification and conversion functions
+
+    strchrisascii/
+	These functions return true
```

Comment:
"Functions defined under this directory return true"

And similarly for each section that isn't a single function like memsory().

Reply:
Done; I think.